librz/core/tui/config.c

```diff
@@ -36,7 +36,8 @@ static void config_visual_hit(RzCore *core, const char *name, int editor) {
 		// XXX: must use config_set () to run callbacks!
 		if (editor) {
 			char *buf = rz_core_editor(core, NULL, node->value);
-			node->value = rz_str_dup(node->value, buf);
+			free(node->value);
+			node->value = rz_str_dup(buf);
```

Comment:
probably you should dup before freeing.
```c
char *tmp = rz_str_dup(buf);
free(node->value);
node->value = tmp;
```
And this probably needs to be done for all the old invocations of `rz_str_dup`

Reply:
Sureee